src/api/app/models/staging/staging_project.rb

```diff
@@ -100,6 +104,11 @@ def unassign_managers_group(managers)
       relationships.find_by(group: managers, role: role).try(:destroy!)
     end
 
+    def failed_status_checks
+      relevant_checks.where(state: Status::Check::FAILED_STATES) +
```

Comment:
[nitpick] @ChrisBr may prefer you use `union` for readability :stuck_out_tongue_closed_eyes:

Reply:
this is used more than once, you should cache it

Reply:
@Ana06 caching: done